sw/host/provisioning/orchestrator/update_remote_db.py

```diff
@@ -0,0 +1,138 @@
+        "test_unlock_token": row[1],
+        "test_exit_token": row[2],
+        "enc_rma_unlock": row[3],
+        "device_ecc_pub_key_x": row[4],
+        "device_ecc_pub_key_y": row[5],
+        "host_ecc_priv_keyfile": row[6],
+        "uds_cert_pub_key_x": row[7],
+        "uds_cert_pub_key_y": row[8],
+        "lc_state": row[9],
+        "sku": row[10],
+        "commit_hash": row[11],
+        "timestamp": row[12]
+    }
+
+
+def _duplicate_prompt(device_id):
+    print(f"Device ID {device_id} already exists in the remote database.")
+    print("Please provide one of the following options:")
+    print("- 'skip': skip this entry and keep it in local DB (default)")
+    print("- 'remove': skip this entry and remove it from local DB")
+    print("- 'overwrite': replace existing entry in remote DB")
+    choice = input("> ")
+
+    if choice in {"skip", "remove", "overwrite"}:
+        return choice
+
+    return "skip"
+
+
+def _delete_entry(local_db, device_id):
+    logging.debug(f"Deleting local record for {device_id}...")
+    local_db.cursor().execute(f"DELETE from {OT_SQL_TABLE_NAME} where device_id = '{device_id}'")
+    local_db.commit()
+
+
+def update_remote_db(project, local_db):
+    logging.info("Uploading local records to remote database")
+    cursor = local_db.cursor()
+
+    res = cursor.execute(f"SELECT * FROM {OT_SQL_TABLE_NAME}")
+    rows = res.fetchall()
+
+    cursor.close()
+
+    remote_db = firestore.Client(project=project)
+    devices_ref = remote_db.collection("devices")
+
+    count = 0
+
+    for row in rows:
+        data = _row_to_dict(row)
+        device_id = data["device_id"]
+        logging.debug(f"Uploading record for {device_id}...")
+
+        query = devices_ref.where(filter=FieldFilter("device_id", "==", device_id))
+        devices = query.get()
+
+        if len(devices) > 0:
+            choice = _duplicate_prompt(device_id)
+            logging.warning(
+                f"Device ID {device_id} already exists in remote database. "
+                f"User chose to {choice}."
+            )
+
+            if choice == "skip":
+                continue
+            elif choice == "remove":
+                _delete_entry(local_db, device_id)
+                continue
+            elif choice == "overwrite":
+                # fall-through to default logic
+                pass
+            else:
+                assert False
+
+        count += 1
+        doc_ref = devices_ref.document(device_id)
+        doc_ref.set(data)
+
+        _delete_entry(local_db, device_id)
+
+    logging.info(f"Successfully uploaded {count} record(s) to remote database")
+
+
+def main():
+    logging.basicConfig(level=logging.INFO)
+
+    parser = argparse.ArgumentParser(
+        prog="Remote Database Update",
+        description="This tool updates the ES provisioning remote database with "
+                    "locally cached device registrations.",
+    )
+    parser.add_argument("db_path",
```

Comment:
Is `db_path` also a required argument?

Reply:
Positional arguments are required by default (and specifying `required` results in an error).